### PR TITLE
Make share results distinguishable if there are more than one with the same display name

### PR DESCRIPTION
**Pull request by juliusknorr**
- Give a context for users that have the same display name in the sharing input and share list
- The context is the email address if set, otherwise it will fallback to the uid

### Keeping current behavior if display names are different

image

### Same display name (first two users have an email set)

image

### Share listing

image

### CODE REVIEW

**Comment by juliushaertl** [comment]
@jancborchardt Could you maybe give me some input here? Having the additional bottom line is not ideal as we usually have user status there now, though it is not yet implemented in the sharing autocomplete dialog.

**Comment by jancborchardt** [comment]
> @jancborchardt Could you maybe give me some input here? Having the additional bottom line is not ideal as we usually have user status there now, though it is not yet implemented in the sharing autocomplete dialog.

@juliushaertl Ah good point – how about putting it directly after the display name? Set as `var(--color-text-maxcontrast)` so it’s not so harsh. Possibly also parentheses around.

**Comment by juliushaertl** [comment]
Yep, sounds good, any idea about if we should show the uid or email? quick summary:
- uid is unique but might be random characters on ldap setup
- email might not be set for the users with the same displayname or might not be unique

**Comment by jancborchardt** [comment]
I would tend towards email since we don’t really use the user id often, so it’s easier to identify people by mail. If it’s not unique, then they are also the same person?

**Comment by juliushaertl** [comment]
>  If it’s not unique, then they are also the same person?

Yep indeed :grin: :+1:

**Comment by juliushaertl** [comment]
Addressed the discussion and updated the first post with details about the implementation, ready for review.

**Review by MorrisJobke — APPROVED** [review]
Code looks good 👍

**Review by ChristophWurst — APPROVED** [review]

**Review by MorrisJobke — DISMISSED** [review]
CI fails.

**Comment by MorrisJobke** [comment]
CI still fails

**Comment by juliushaertl** [comment]
OK, tests should be happy now finally. 

The formatShare test data provider is a beast with its 600 lines :imp:

**Comment by ChristophWurst** [comment]
needs rebuild, no?

**Comment by juliushaertl** [comment]
Yes, but I usually wait until reviewed, to avoid unneded rebuilding over and over again if something else get's merged ;)

**Comment by tobiasKaminsky** [comment]
@juliushaertl @juliushaertl isn't this something that also clients should handle/show?
Why is there no ticket/info about that…Or did I missed something? In that case, sorry, of course.

**Comment by jancborchardt** [comment]
> @juliushaertl @juliushaertl isn't this something that also clients should handle/show?
> Why is there no ticket/info about that…Or did I missed something? In that case, sorry, of course.

Yep, good point – we needed to decide how we do it first.

So basically the only change now is that we show the email address in parentheses behind the display name, in `var(--color-text-maxcontrast)`. Right @juliushaertl?

**Comment by juliushaertl** [comment]
> isn't this something that also clients should handle/show?
> Why is there no ticket/info about that…Or did I missed something? In that case, sorry, of course.

Yes definitely, sorry I should have pinged or created an overview for that.

> So basically the only change now is that we show the email address in parentheses behind the display name, in var(--color-text-maxcontrast). Right @juliushaertl?

Yes, though on the multiselect input the description (email to make the entry unique) would currently overwrite the status, but that would be something to fix on the vue components:

https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/Multiselect/AvatarSelectOption.vue#L77-L85

**Comment by juliushaertl** [comment]
@jancborchardt Good from your side then as a first step? We can always iterate over it with adding `(guest)` for example in a later step, but this is more a bugfix since otherwise users could not distinguish between other users with the same name when sharing.

**Comment by MorrisJobke** [comment]
conflicts :(

**Review by jancborchardt — APPROVED** [review]
Great solution design-wise! :)

**Comment by juliushaertl** [comment]
Rebased and rebuilt.

**Comment by juliushaertl** [comment]
Resolved again

File: apps/files_sharing/src/models/Share.js [hunk 1]
```diff
@@ -150,6 +150,10 @@ export default class Share {
 			|| this.#share.share_with
 	}
 
+	get shareWithDisplayNameUnique() {
```
Comment (skjnldsv) [inline apps/files_sharing/src/models/Share.js:153]: Tss tss tss @juliushaertl , no documentation? :grin: :see_no_evil:

**Pull request merged**